src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -1051,6 +1051,21 @@ private bool TryAddInternal(TKey key, int? nullableHashcode, TValue value, bool
         private static void ThrowKeyNotFoundException(TKey key) =>
             throw new KeyNotFoundException(SR.Format(SR.Arg_KeyNotFoundWithKey, key.ToString()));
 
+        public IEqualityComparer<TKey> Comparer
+        {
+            get
+            {
+                if (typeof(TKey) == typeof(string))
+                {
+                    return (IEqualityComparer<TKey>)IInternalStringEqualityComparer.GetUnderlyingEqualityComparer((IEqualityComparer<string?>?)_comparer);
+                }
+                else
+                {
+                    return _comparer ?? EqualityComparer<TKey>.Default;
+                }
+            }
```

Comment:
The entire implementation should just be:
```C#
public IEqualityComparer<TKey> Comparer => _comparer ?? _defaultComparer;
```

Reply:
Shouldn't this be consistent with [what the regular dictionary is doing](https://github.com/dotnet/runtime/blob/2e0ae35947d29c8be6357f0cb95205921fbfa28d/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs#L172-L179)?

Reply:
No.  Dictionary upgrades its default string comparer from a (corelib internal) non-randomized comparer to a randomized comparer, hence it needing special logic in Comparer to compensate and hide the implementation detail.  ConcurrentDictionary has no such implementation.